bsmetadata/metadata_utils.py

```diff
@@ -182,15 +182,26 @@ def _collate_metadata(metadata_list: List[dict], cfg: MetadataConfig):
 
         assert metadata_node.relative_start_pos not in metadata_dict_idx[metadata_node.char_start_idx]
         assert metadata_node.relative_end_pos not in metadata_dict_idx[metadata_node.char_end_idx]
-
-        metadata_dict_idx[metadata_node.char_start_idx][metadata_node.relative_start_pos] = start_text
-        metadata_dict_idx[metadata_node.char_end_idx][metadata_node.relative_end_pos] = end_text
+        
+        if start_text:
+            metadata_dict_idx[metadata_node.char_start_idx][metadata_node.relative_start_pos] = start_text
+        if end_text:
+            metadata_dict_idx[metadata_node.char_end_idx][metadata_node.relative_end_pos] = end_text
 
     for absolute_idx, value in metadata_dict_idx.items():
         pos_sorted = sorted(list(value.keys()))
         local_metadata = ""
         for pos in pos_sorted:
             local_metadata += metadata_dict_idx[absolute_idx][pos]
+
+        # We add here a local special token if needed around the metadata list of a type if needed
+        if local_metadata and metadata_list[0]["key"] in cfg.local_metadata_special_token_start:
```

Comment:
@manandey I think we should also check if `cfg.local_metadata_special_token_start` is non null before iterating through it. Same for `special_token_end`